ports/geckolib/glue.rs

```diff
@@ -1377,6 +1377,29 @@ pub extern "C" fn Servo_DeclarationBlock_SerializeOneValue(
 }
 
 #[no_mangle]
+pub extern "C" fn Servo_SerializeFontValueForCanvas(
+    declarations: RawServoDeclarationBlockBorrowed,
+    buffer: *mut nsAString) {
+    use style::properties::ShorthandId;
+    use style::properties::shorthands::font;
+
+    read_locked_arc(declarations, |decls: &PropertyDeclarationBlock| {
+        debug_assert!(decls.declarations().iter().all(|decl| decl.0.shorthands().contains(&ShorthandId::Font)),
+                      "Should have only font shorthand");
+        if !decls.declarations().iter().all(|decl| decl.0.shorthands().contains(&ShorthandId::Font)) {
```

Comment:
Is this failing in any case? Shouldn't it be easier to d something like:

```rust
let longhands = match font::LonghandsToSerialize::from_iter(decls.declarations_iter()) {
    Some(l) => l,
    None => {
        warn!("Unexpected property!");
        return;
    }
}
// ...
```

That'd avoid walking over the declarations twice, I think.